src/olympia/scanners/models.py

```diff
@@ -402,17 +418,25 @@ def completion_rate(self):
 
 
 class ScannerQueryResult(AbstractScannerResult):
-    # Has to be overridden, because the parent refers to ScannerMatch.
-    matched_rules = models.ManyToManyField(
-        'ScannerQueryRule', through='ScannerQueryMatch', related_name='results'
+    # Note: ScannerResult uses a M2M called 'matched_rules', but here we don't
+    # need a M2M because there will always be a single rule for each result, so
+    # we have a single FK called 'matched_rule' (singular).
+    matched_rule = models.ForeignKey(
+        ScannerQueryRule, on_delete=models.CASCADE, related_name='results'
     )
```

Comment:
This is the key change - there is a lot of code being moved around from the abstract classes to the more specific classes in this PR, but this change is the most important part, changing the M2M to a FK.